src/ProjectTemplates/test/ProjectTemplates.Tests.Common/Infrastructure/GenerateTestProps.targets

```
@@ -0,0 +1,94 @@
+<Project>
+  <Target
+    Name="GenerateTestProps"
+    BeforeTargets="CoreCompile"
+    DependsOnTargets="PrepareForTest"
+    Condition="$(DesignTimeBuild) != true">
+    <PropertyGroup>
+      <PropsProperties>
+        RestoreSources=$([MSBuild]::Escape("$(RestoreSources);$(ArtifactsShippingPackagesDir);$(ArtifactsNonShippingPackagesDir)"));
+        MicrosoftNETCoreAppPackageVersion=$(MicrosoftNETCoreAppPackageVersion);
+        MicrosoftNETCorePlatformsPackageVersion=$(MicrosoftNETCorePlatformsPackageVersion);
+        MicrosoftNETSdkRazorPackageVersion=$(MicrosoftNETSdkRazorPackageVersion);
+        MicrosoftAspNetCoreAppPackageVersion=$(SharedFxVersion);
+        SupportedRuntimeIdentifiers=$(SupportedRuntimeIdentifiers);
+      </PropsProperties>
+    </PropertyGroup>
+
+    <Sdk_GenerateFileFromTemplate
+      TemplateFile="$(MSBuildThisFileDirectory)\TemplateTests.props.in"
```

```suggestion
      TemplateFile="$(MSBuildThisFileDirectory)TemplateTests.props.in"
```

MSBuild coding convention - MSBuildThisFileDirectory already ends in a trailing slash.